bfabric/bfabric2.py

```diff
@@ -161,21 +170,32 @@ def read(self, endpoint: str, obj: dict, max_results: Optional[int] = 100, readi
             n_pages_trg = min(n_pages, div_int_ceil(max_results, BFABRIC_QUERY_LIMIT))
 
         # NOTE: Page numbering starts at 1
-        response_list = response[endpoint]
+        response_items = response[endpoint]
+        errors = []
         for i_page in range(2, n_pages_trg + 1):
             print('-- reading page', i_page, 'of', n_pages)
-            response_list += self._read_method(readid, endpoint, obj, page=i_page, **kwargs)[endpoint]
+            response = self._read_method(readid, endpoint, obj, page=i_page, **kwargs)
+            errors += self._get_response_errors(response)
```

Comment:
To clarify: The _get_response_errors is always called on the "raw" response from the engine now, i.e. without extracting `[endpoint]` specific list of results yet.

I agree we should investigate this further. My intuition would be that reads specifically either pass or fail as a whole, because if you request an invalid field, somewhere in the stack of the SOAP communication it might be rejected already. However maybe your login expires between pages or there is some other type of error that leads to only some pages not available. I guess we shouldn't try to figure out if all of the requested items were found, since it's not possible to do this in general.

However the point you brought up about one of several writes failing, we really need to check what the API does there and would have to handle it appropriately if we want to support the multi-write.

Reply:
So can we make a few integration tests to test things that we do know already. For example:
* Deleting non-existing entries
* Saving onto endpoints we are not allowed to save to
Or would you prefer to test at a later stage?